### PR TITLE
ipam/allocator: Fix nil check on node CIDR

### PR DESCRIPTION
**Pull request by christarazi**
See commit msgs.

Fixes: ef6ecbdb61 ("add error log When ipam allocate nodecidr failure")
Fixes: https://github.com/cilium/cilium/pull/13299

```release-note
Fix bug in cluster-pool IPAM mode where the user is never alerted of a node CIDR allocation failure
```

### CODE REVIEW

**Comment by christarazi** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by pchaigno** [comment]
retest-gke

**Comment by christarazi** [comment]
retest-gke

**Comment by aanm** [comment]
retest-net-next

**Comment by christarazi** [comment]
Marking ready to merge as we have code owner approval and all CI passing.

**Pull request merged**
